core/input/input_map.cpp

```diff
@@ -667,6 +668,11 @@ const OrderedHashMap<String, List<Ref<InputEvent>>> &InputMap::get_builtins() {
 	inputs.push_back(InputEventKey::create_reference(KEY_MENU));
 	default_builtin_cache.insert("ui_menu", inputs);
 
+	inputs = List<Ref<InputEvent>>();
+	inputs.push_back(InputEventKey::create_reference(KEY_ENTER));
+	inputs.push_back(InputEventKey::create_reference(KEY_ENTER | KEY_MASK_SHIFT));
+	default_builtin_cache.insert("ui_text_enter", inputs);
+
```

Comment:
It seems wrong to have Shift+Enter for the "Text Entered" shortcut...

I think a better solution could be to only have `KEY_ENTER` and `KEY_KP_ENTER` assigned to this, and in the `is_action` check in LineEdit do a non-exact check (i.e. second param is `false`). This would allow for Shift-Enter to trigger the signal, and then the current logic downstream (checking if Shift is pressed) would still work as it currently does.

Reply:
That is good to know! I'm still learning how godot works, but I should have looked more closely at how `is_action` works. Will be updated in latest push